core/src/main/java/org/apache/cxf/common/util/PackageUtils.java

```diff
@@ -56,7 +57,9 @@ public static String getSharedPackageName(List<Class<?>> classes) {
         List<List<String>> lParts = new ArrayList<List<String>>(classes.size());
         List<String> currentParts = new ArrayList<>();
         for (Class<?> cls : classes) {
-            lParts.add(StringUtils.getParts(getPackageName(cls), "\\."));
+            if (!Proxy.isProxyClass(cls)) {
```

Comment:
In this case, should we get the real class using `getProxyClass` and extract the package from it?